GVFS/GVFS.Common/ProductUpgraderInfo.cs

```diff
@@ -21,19 +21,29 @@ public static string CurrentGVFSVersion()
             return ProcessHelper.GetCurrentProcessVersion();
         }
 
-        public static string GetUpgradesDirectoryPath()
-        {
-            return GVFSPlatform.Instance.GetDataRootForGVFSComponent(UpgradeDirectoryName);
+        public static string GetUpgradeApplicationDirectory()
+        {
+            return Path.Combine(GVFSPlatform.Instance.GetUpgradeProtectedDataDirectory(), ToolsDirectory);
+        }
+
+        public static string GetUpgradeProtectedDataDirectory()
+        {
+            return GVFSPlatform.Instance.GetUpgradeProtectedDataDirectory();
+        }
+
+        public static string GetUpgradeBasicDataDirectory()
+        {
+            return GVFSPlatform.Instance.GetUpgradeNonProtectedDataDirectory();
         }

```

Comment:
Can we use the same name and terminology throughout the code.  Here `Protected` matches but `Basic == NonProtected`?  I was also thinking that the terms should be `Restricted` and `User` especially since `Basic` is a little to generic and doesn't really describe it.

Reply:
As the concept of base `User` writable directory does not exist across platforms at this time, I removed `Basic` for now. I still have `Protected`, but I like `Restricted` as well.